app/core/models/ocelot/LabelCache.scala

```diff
@@ -20,31 +20,55 @@ import play.api.i18n.Lang
 
 trait Labels {
   def value(name: String): Option[String]
+  def valueAsList(name: String): Option[List[String]]
   def displayValue(name: String)(implicit lang: Lang): Option[String]
   def update(name: String, english: String): Labels
   def update(name: String, english: String, welsh: String): Labels
+  def updateList(name: String, english: List[String]): Labels
+  def updateList(name: String, english: List[String], welsh: List[String]): Labels
   def updatedLabels: Map[String, Label]
   def labelMap:Map[String, Label]
   def flush(): Labels
 }
 
 private class LabelCacheImpl(labels: Map[String, Label] = Map(), cache: Map[String, Label] = Map()) extends Labels {
-  def value(name: String): Option[String] = label(name).map(_.english.getOrElse(""))
+  def value(name: String): Option[String] = label(name).fold[Option[String]](None)
+    {case s:ScalarLabel => Some(if(s.english.nonEmpty)s.english.head else "") case _:ListLabel => None}
+  def valueAsList(name: String): Option[List[String]] = label(name).fold[Option[List[String]]](None)
+    {case l:ListLabel => Some(l.english) case _:ScalarLabel => None}
   def displayValue(name: String)(implicit lang: Lang): Option[String] = label(name).map{lbl =>
     lang.code match {
-      case "en" => lbl.english.getOrElse("")
-      case "cy" => lbl.welsh.fold(lbl.english.getOrElse(""))(cy => cy)
+      case "en" => lbl.english.mkString(",")
+      case "cy" => lbl.welsh.mkString(",")
     }
   }
-  def update(name: String, english: String): Labels = new LabelCacheImpl(labels, updateOrAddLabel(name, english))
-  def update(name: String, english: String, welsh: String): Labels = new LabelCacheImpl(labels, updateOrAddLabel(name, english, Some(welsh)))
+  def update(name: String, english: String): Labels = new LabelCacheImpl(labels, updateOrAddScalarLabel(name, english))
+  def update(name: String, english: String, welsh: String): Labels = new LabelCacheImpl(labels, updateOrAddScalarLabel(name, english, welsh))
+  def updateList(name: String, english: List[String]): Labels = new LabelCacheImpl(labels, updateOrAddListLabel(name, english))
+  def updateList(name: String, english: List[String], welsh: List[String]): Labels = new LabelCacheImpl(labels, updateOrAddListLabel(name, english, welsh))
   def updatedLabels: Map[String, Label] = cache
   def labelMap:Map[String, Label] = labels
   def flush(): Labels = new LabelCacheImpl(labels ++ cache.toList, Map())
 
   private def label(name: String): Option[Label] = cache.get(name).fold(labels.get(name))(Some(_))
-  private def updateOrAddLabel(name: String, english: String, welsh: Option[String] = None): Map[String, Label] =
-    cache + (name -> cache.get(name).fold[Label](Label(name, Some(english), welsh))(l => Label(l.name, Some(english), welsh)))
+
+  private def updateOrAddScalarLabel(name: String, english: String): Map[String, Label] = {
```

Comment:
think this and following private fns could be merged into one fn using default arg

Reply:
Fixed